### PR TITLE
Add ACL token to Consul module 

### PR DESCRIPTION
**Pull request by rtx3**
### What does this PR do?
1. Add ACL token to consul module when consul set default ACL rule to…deny.
2. Add node meta to function catalog_register.
3. Fix a bug when function catalog_register return true but actually not registered.

### What issues does this PR fix or reference?
When using consul with ACL requirement

### Tests written?
No
Tested with below conditions: 
consul: 0.9.2
salt: 2017.7.1
function:
catalog_register
catalog_deregister
agent_check_register
agent_check_pass
Not enough time to test all functions, but I believe all functions should work.
Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

### CODE REVIEW

File: salt/modules/consul.py [hunk 1]
```diff
@@ -200,20 +214,17 @@ def get(consul_url=None, key=None, recurse=False, decode=False, raw=False):
         query_params['raw'] = True
     ret = _query(consul_url=consul_url,
                  function=function,
+                 token=token,
                  query_params=query_params)
 
     if ret['res']:
         if decode:
             for item in ret['data']:
-                if item['Value'] != None:
```
Comment (vladislavPV) [inline salt/modules/consul.py]: you have to put that back , other way an empty key will cause error

**Comment by cachedout** [comment]
Hi @rtx3. Thanks for this. There are a number of lint errors, however. Would you mind having a look and fixing them up? Thanks.

https://jenkins.saltstack.com/job/PR/job/salt-pr-lint-n/14440/violations/file/salt/modules/consul.py/

**Comment by garethgreenaway** [comment]
Looks like quite a bit of Lint errors.  Also agree with @vladislavPV 's comments on the empty key changes.  Lastly, adding the new `token` argument to all the functions, usually best practices to add any new arguments to the end to avoid issues with existing calls to those functions.

**Comment by rtx3** [comment]
@cachedout Sorry for that, I've fixed my code.
 @garethgreenaway @vladislavPV 
Thanks for the review. I agree with the empty key , my mistake. However, adding a new function with  new arguments may not be the best option. Since all function would call _query to call Consul API, if I added a new _query function in this module would make  a lot of confusion  for others.

**Review by garethgreenaway — APPROVED** [review]

File: salt/modules/consul.py [hunk 2]
```diff
@@ -101,7 +114,7 @@ def _query(function,
     return ret
 
 
-def list_(consul_url=None, key=None, **kwargs):
+def list_(consul_url=None, token=None, key=None, **kwargs):
```
Comment (rallytime) [inline salt/modules/consul.py]: Can you move the addition of the `token` option after `key`? That way we can avoid an API change for other modules calling out to this function.

File: salt/modules/consul.py [hunk 3]
```diff
@@ -140,14 +154,15 @@ def list_(consul_url=None, key=None, **kwargs):
...
+    query_params['separator'] = '/'
     ret = _query(consul_url=consul_url,
                  function=function,
+                 token=token,
                  query_params=query_params)
     return ret
 
 
-def get(consul_url=None, key=None, recurse=False, decode=False, raw=False):
+def get(consul_url=None, key=None, token=None, recurse=False, decode=False, raw=False):
```
Comment (rallytime) [inline salt/modules/consul.py]: Same comment here. This new kwarg needs to be added to the end of the kwarg list.

File: salt/modules/consul.py [hunk 4]
```diff
@@ -209,11 +225,10 @@ def get(consul_url=None, key=None, recurse=False, decode=False, raw=False):
                     item['Value'] = base64.b64decode(item['Value'])
                 else:
                     item['Value'] = ""
-
     return ret
 
 
-def put(consul_url=None, key=None, value=None, **kwargs):
+def put(consul_url=None, token=None, key=None, value=None, **kwargs):
```
Comment (rallytime) [inline salt/modules/consul.py]: Same comment here.

There are quite a few functions below where the `token` option needs to be moved. Please update this anywhere else in this PR below as well.

**Review by rallytime — CHANGES_REQUESTED** [review]
Thanks for this change @rtx3 and welcome!

There are a couple of changes that need to be taken care of before we can get this in, largely around where you're adding the `token` kwarg to various functions. Those references need to be moved to the end of each kwarg list so we can avoid an API change.

Also, with the addition of this option, the documentation also needs to be updated for all of the functions adding the `token` option.

Once those updates are made, we can re-review this. Thank you!

**Comment by cachedout** [comment]
@rtx3 Did you see the above review comments and requests for changes? We would love to get this in if you can come back to it. Thanks.

**Comment by 1ricardo** [comment]
Hello,

I would like ask @cachedout , what is status of that feature. For me module.consul.put stopped worked, when we implemented the ACL in consul, from security reason.

Best Regards

**Pull request merged**